app/controllers/application_controller.rb

```diff
@@ -929,6 +930,29 @@ def should_enforce_2fa?
       !current_user.has_any_second_factor_methods_enabled?
   end
 
+  def redirect_to_profile_if_required
+    return if request.format.json? && is_api?
+    return if !current_user
+    return if !current_user.needs_required_fields_check?
+
+    if current_user.populated_required_custom_fields?
+      current_user.bump_required_fields_version
+      return
+    end
+
+    redirect_path = path("/u/#{current_user.encoded_username}/preferences/profile")
+    second_factor_path = path("/u/#{current_user.encoded_username}/preferences/second-factor")
+    allowed_paths = [redirect_path, second_factor_path, "/admin"]
```

Comment:
Do we need to wrap `/admin` with `path()`?

Reply:
I suspect we might ... subfolder thingy?